### ISSUE TITLE
Comments: Display a Note

### ISSUE THREAD

**Issue by jayair** [AnomalyInnovations/serverless-stack-com#53]

**Comment by mikhaelbendavid** [comment]
When I click on a note, the link doesn't change to the noteid, instead I get to localhost:3000/undefined - I am very perplexed.

My getNote() should be working, as it is:
getNote() {
    return invokeApig({ path: `/notes/${this.props.match.params.id}` }, this.props.userToken);
  }

If I control copy and paste localhost:3000/noteIdhere by looking it up manually on my DynamoDb, then it pops up. I can also create new notes from local server, but I'm not able to automatically shift the new URL to correspond to the notesid. 

I went back to look at how invokeApig is set up in /libs/awsLib and there doesn't seem to be any issues - here is my AppliedRoute in Routes.JS
<AppliedRoute path="/notes/:id" exact component={Notes} props={childProps} />

Any thoughts? When inspecting, my notes on the home page show up as Objects, though when rendering the URL just changes to undefined. I've been stuck on this for a few days, it must be something really simple and basic that I'm missing - thanks for your patience with a newbie. Here's my repo: https://github.com/mikhaelbendavid/serverlessblog/blob/master/notes-app-client/src/containers/Notes.js

**Comment by AndrewWrightNZ** [comment]
Hey I can view all notes in the list view on Home.js but when I click on a note I get the error:

TypeError: Failed to fetch

{message: "Missing Authentication Token"}
message
:
"Missing Authentication Token"

Any ideas?

**Comment by jayair** [comment]
@AndrewWrightNZ That's weird that it's looking for the token. Can I just do a sanity check and look at your `serverless.yml` for the backend?

**Comment by AndrewWrightNZ** [comment]
@jayair Yeah sure, https://github.com/AndrewWrightNZ/notes-app-api/blob/master/serverless.yml

**Comment by jayair** [comment]
@AndrewWrightNZ Yeah that looks okay.

In a previous version of the tutorial we were sending the `userToken` in each of the `invokeApig` calls. Can I see your `Home.js`?

**Comment by AndrewWrightNZ** [comment]
Yeah - https://github.com/AndrewWrightNZ/notes-app-client/blob/master/src/containers/Home.js

From a quick google search it may be AWS config related?

**Comment by jayair** [comment]
@AndrewWrightNZ Not sure if this is the right version but I tried your repo and the `invokeApig` call in `Home.js` is making a request to the wrong API endpoint? And there is no `Notes.js`? If you push the updated version I'll take a look.

**Comment by AndrewWrightNZ** [comment]
@jayair awesome, just reviewed all of the routes and its working 💯

**Comment by tommedema** [comment]
Is there any reason why some async functions are not defined as async. Is this deliberate?

I.e.

```
getNote() {
    return invokeApig({ path: `/notes/${this.props.match.params.id}` });
}
```

instead of

```
async getNote() {
    return await invokeApig({ path: `/notes/${this.props.match.params.id}` });
}
```

**Comment by jayair** [comment]
@tommedema Not all functions that return Promises need to be marked as async. The better way to think about this is; mark a function as async if you need to use await, as that is the purpose of the async keyword.

There are other advantages to doing things this way, especially if you intend to act on those promises differently. But in the case we aren't doing anything special with the promise, so this would be more of a style choice.

**Comment by eldadmel** [comment]
My note.noteId is always "undefined". Any idea what I did wrong?

**Comment by jayair** [comment]
@eldadmel You can try to `console.log(results)` to see what you are getting back from the API.

**Comment by mbahfauz** [comment]
He Jay, may I know how to render images on list notes? I had tried a different way, but still stuck with it.
Because I only got a database attachment text.

**Comment by mbahfauz** [comment]
Here is my code


`import React, { Component } from "react";
import { PageHeader, ListGroup, ListGroupItem } from "react-bootstrap";
import "../Home.css";
import { API } from "aws-amplify";
import { Link } from "react-router-dom"

export default class Home extends Component {
    constructor(props) {
        super(props);

        this.state = {
            isLoading: true,
            coins: []
        };
    }





    async componentDidMount() {
        if (!this.props.isAuthenticated) {
            return;
        }

        try {
            const coins = await this.coins();
            this.setState({ coins });
        } catch (e) {
            alert(e);
        }

        this.setState({ isLoading: false });
    }

    coins() {
        return API.get("notes", "/coins");
    }


    renderCoinsList(coins) {
        return [{}].concat(coins).map(
            (coin, i) =>
                i !== 0
                    ?
                    <div >

                        <div className="m-widget5">
                            <div className="m-widget5__item"  key={coin.coinId} style={{borderBottom: "0.07rem solid #ebedf2"}}>
                                <div className="m-widget5__pic">
                                    <img className="m-widget7__img" src="/assets/app/media/img/products/product6.jpg" alt="" />
                                </div>
                                <div className="m-widget5__stats1">
                                    <span className="m-widget5__number">{coin.namecoin}</span><br/>

                                </div>
                                <div className="m-widget5__stats1">
                                    <span className="m-widget4__title">{coin.currency}</span><br/>

                                </div>
                                <div className="m-widget5__stats1">
                                    <span className="m-widget4__title">{coin.pricebtc}</span><br/>

                                </div>
                                <div className="m-widget5__stats1">
                                    <span className="m-widget4__titler">{coin.priceusd}</span><br/>

                                </div>
                                <div className="m-widget5__stats1">
                                    <span className="m-widget4__title">{coin.currentblock}</span><br/>

                                </div>
                                <div className="m-widget5__stats1">
                                    <span className="m-widget4__title">more info</span><br/>

                                </div>

                            </div>


                        </div>





                    </div>
                    : <div/>
        );
    }

    handleNoteClick = event => {
        event.preventDefault();
        this.props.history.push(event.currentTarget.getAttribute("href"));
    }

    renderLander() {
        return (
            <div className="lander">
                <h1>Scratch</h1>
                <p>A simple note taking app</p>
            </div>
        );
    }

    renderCoins() {
        return (
            <div className="notes">
                <ListGroup>
                    {!this.state.isLoading && this.renderCoinsList(this.state.coins)}
                </ListGroup>
            </div>
        );
    }

    render() {
        return (
            <div className="Home">





                <div className="m-portlet m-portlet--full-height ">
                    <div className="m-portlet__head">
                        <div className="m-portlet__head-caption">
                            <div className="m-portlet__head-title">
                                <h3 className="m-portlet__head-text">
                                    Coins List
                                </h3>
                            </div>
                        </div>
                        <div className="m-portlet__head-tools">
                            <ul className="nav nav-pills nav-pills--brand m-nav-pills--align-right m-nav-pills--btn-pill m-nav-pills--btn-sm" role="tablist">
                                <li className="nav-item m-tabs__item">
                                    <Link className="nav-link m-tabs__link active"  to="/coins/new" >
                                        Add Coin
                                    </Link>
                                </li>

                            </ul>
                        </div>
                    </div>
                    <div className="m-portlet__body">

                        <div className="tab-content">
                            <div className="tab-pane active" id="m_widget5_tab1_content" aria-expanded="true">
                                <div className="m-widget5">
                                    <div className="m-widget5__item"   style={{borderBottom: "0.07rem solid #ebedf2"}}>
                                        <div className="m-widget5__stats1" style={{textAlign: 'left'}}>
                                            <span className="m-widget5__number">Logo</span><br/>

                                        </div>
                                        <div className="m-widget5__stats1">
                                            <span className="m-widget5__number">Name</span><br/>

                                        </div>
                                        <div className="m-widget5__stats1">
                                            <span className="m-widget5__number">currency</span><br/>

                                        </div>
                                        <div className="m-widget5__stats1">
                                            <span className="m-widget5__number">price btc</span><br/>

                                        </div>
                                        <div className="m-widget5__stats1">
                                            <span className="m-widget5__number">price usd</span><br/>

                                        </div>
                                        <div className="m-widget5__stats1">
                                            <span className="m-widget5__number">current block</span><br/>

                                        </div>
                                        <div className="m-widget5__stats1">
                                            <span className="m-widget5__number">etc</span><br/>

                                        </div>

                                    </div>


                                </div>



                                {this.props.isAuthenticated ? this.renderCoins() : this.renderLander()}


                            </div>


                        </div>

                    </div>
                </div>





            </div>
        );
    }
}`

**Comment by jayair** [comment]
@mbahfauz Here is some info on handling images in Create React App - https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#adding-images-fonts-and-files
